packages/mobile/src/tokens/selectors.ts

```diff
@@ -102,10 +102,15 @@ export const defaultTokenToSendSelector = createSelector(
 )
 
 export const totalTokenBalanceSelector = createSelector(
-  [tokensWithUsdValueSelector, localCurrencyExchangeRatesSelector, tokenFetchErrorSelector],
-  (tokensWithUsdValue, exchangeRate, tokenFetchError) => {
+  [
+    tokensListSelector,
+    tokensWithUsdValueSelector,
+    localCurrencyExchangeRatesSelector,
+    tokenFetchErrorSelector,
+  ],
+  (tokensList, tokensWithUsdValue, exchangeRate, tokenFetchError) => {
     const usdRate = exchangeRate[Currency.Dollar]
-    if (!usdRate) {
+    if (!usdRate || Object.keys(tokensList).length === 0) {
```

Comment:
i think it makes more sense for this selector to return null rather than 0 if the token list has not yet loaded, but also without this we cannot differentiate between an account funded event and a user who is restoring their account (going from 0 to a non-zero balance)